client/src/chat/MessageBubble.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState } from "react";
+import { relTime } from "./chatUtils";
+import { ReactionList } from "./ReactionList";
+import { EmojiPicker } from "./EmojiPicker";
+import { EmojiIcon } from "./Icons";
+
+export function MessageBubble({
+  message,
+  isSelf,
+  showNickname,
+  showTitle,
+  reactionEmojisAvailable,
+  reactToSelf,
+  onAddReaction,
+  onRemoveReaction,
+  players,
+  currentPlayerPosition,
+}) {
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const { text, sender, createdAt, reactions } = message;
+
+  let { avatar } = sender;
+  if (!avatar) {
+    avatar = `https://avatars.dicebear.com/api/identicon/${sender.id}.svg`;
+  }
+
+  const avatarImage = avatar.startsWith("http") ? (
+    <img
+      className="inline-block h-8 w-8 rounded-full flex-shrink-0"
+      src={avatar}
+      alt={sender.id}
+    />
+  ) : (
+    <div className="inline-block h-8 w-8 rounded-full flex-shrink-0">
+      {avatar}
+    </div>
+  );
+
+  const canReact =
+    reactionEmojisAvailable &&
+    reactionEmojisAvailable.length > 0 &&
+    (reactToSelf || !isSelf);
+
+  const handleEmojiSelect = (emoji) => {
+    if (onAddReaction) {
+      onAddReaction(message.id, emoji);
+    }
+    setShowEmojiPicker(false);
+  };
+
+  if (isSelf) {
+    // Right-aligned blue bubble for self messages
+    return (
+      <div className="flex justify-end items-start my-2 px-2">
+        <div
+          className="relative max-w-[75%]"
+          onMouseEnter={() => setIsHovered(true)}
+          onMouseLeave={() => {
+            setIsHovered(false);
+            setShowEmojiPicker(false);
+          }}
+        >
+          <div className="flex flex-col items-end gap-1">
+            <div className="bg-blue-500 text-white rounded-2xl px-4 py-2 break-words">
+              <p className="text-sm leading-relaxed">{text}</p>
```

Comment:
lets make sure we have even spacing on top and bottom - so vertically center the text in the bubble. Also, make the text white for the self-messages.